### DISCUSSION TITLE
XTouch display support

### DISCUSSION THREAD

**Discussion by ghost** [rsjaffe/MIDI2LR#657] [Ideas]
I recently bought a Behringer XTouch.
Does MIDI2LR support the displays in the XTouch? By the displays I mean the small displays below each encoder knob (color and text) and the digits on the right LED panel.
It would be nice if the displays could show preset names and the LED panel could show the EXIM info (shutter speed, f-number, ISO and maybe focal length)

**Comment by rsjaffe** [comment]
No. Different controllers have different protocols for their displays. Doing this would be a huge project, which I'm not up for.

**Reply by anwright** [reply]
I came here looking for basically the same thing, except for a PreSonus FaderPort 16.  I'm actually looking for something that I think would be much less of a project, though: the ability to just send arbitrary SysEx messages upon loading a profile.  I have hand-crafted a SysEx file that will set the correct values on all the scribble strips (displays), but have no way to update it whenever a profile loads.  If MIDI2LR could either look for a sidecar .syx file (matching the .xml file name, or defined within the profile), or allow embedding of the raw sysex data in the profile itself, that would be sufficient for me.  I'd create a few profiles, a few sysex files to set the text appropriately per-slider, and whenever the profile loaded, I'd have correct labels for everything.

**Reply by rsjaffe** [reply]
Could you post a copy of the syx file? That’ll help me understand the amount of work needed to do this.

**Reply by anwright** [reply]
Apologies for the delay, I intended to add a sample of setting LED color values as well, but discovered that the FP16's LEDs are controlled with standard MIDI messages and not SysEx.  [Here's a sample .syx file](https://github.com/rsjaffe/MIDI2LR/files/9136450/fp16.syx.gz), which sets scribble strips for channels 0-14 to samples of what I'd actually use them for, and sets 15 to just one example of a different format.  On this particular device, there are quite a few options to choose from.  The format is described starting on page 79 of the [owner's manual](https://pae-web.presonusmusic.com/downloads/products/pdf/FaderPort_16-8_OwnersManual_V5_EN_28092021.pdf).
